File: frontend/src/features/auth/services/authService.ts
import api from '@/lib/api'
import type { LoginRequest, LoginResponse } from '../types/auth.types'

export const authService = {
  login: (data: LoginRequest): Promise<LoginResponse> =>
    api.post<LoginResponse>('/auth/login', data).then((r) => r.data),
  me: (): Promise<LoginResponse> => api.get<LoginResponse>('/auth/me').then((r) => r.data),
  refresh: (): Promise<LoginResponse> =>
    api.post<LoginResponse>('/auth/refresh').then((r) => r.data),
  logout: (): Promise<void> => api.post('/auth/logout').then(() => undefined),
}

File: backend/src/main/java/com/solutionti/usuarios/service/AuthService.java
package com.solutionti.usuarios.service;

import com.solutionti.usuarios.dto.request.LoginRequest;
import com.solutionti.usuarios.dto.response.LoginResponse;
import com.solutionti.usuarios.service.auth.AuthSessionResult;

public interface AuthService {

    AuthSessionResult login(LoginRequest request);

    AuthSessionResult refresh(String refreshToken);

    void logout(String refreshToken);

    LoginResponse me();
}
